core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -1045,14 +1045,30 @@ private AnnotatedValueResolver build() {
                     defaultValue = null;
                 }
             } else {
-                shouldExist = !shouldWrapValueAsOptional;
                 // Set the default value to null if it was not specified.
                 defaultValue = null;
+
+                if (shouldWrapValueAsOptional) {
+                    shouldExist = false;
+                } else {
+                    // Allow `null` if annotated with `@Nullable`.
+                    boolean isNonNull = true;
+                    for (Annotation a : annotatedElement.getAnnotations()) {
+                        final String annotationTypeName = a.annotationType().getName();
+                        System.err.println(annotationTypeName);
+                        if (annotationTypeName.endsWith(".Nullable")) {
```

Comment:
Could use `Class.getSimleName()`? 
```java
"Nullable".equals(a.annoatation.getSimpleName())
```

Reply:
`getSimpleName()` creates a new string, so I wanted to avoid unnecessary overhead.

Reply:
Good to know. 🙇‍♂️
I investigated the related issues. The `simpleName` is cached since Java 11. But it looks pretty expensive.
https://stackoverflow.com/a/57630221/1736581
https://bugs.openjdk.java.net/browse/JDK-8187123